### PR TITLE
Add a global CSS rule for extending pages height and avoid double scrollbar

### PR DESCRIPTION
**Pull request by KKoukiou**
Remove all per page CSS for this purpose.

The single visual difference by this is that loading page spinner is in
the center of the page now, instead of the top that it was before.
This is the expected behavior.

Please review carefully :) Prone to :bug:

### CODE REVIEW

**Comment by garrett** [comment]
This is cool! Thanks for changes. It seems fine with a quick glance at the PR.

I'll build and test locally and look through the diff more closely.

File: pkg/lib/page.scss [hunk 1]
```diff
@@ -549,3 +549,11 @@ input[type=number]:not(.pf-c-form-control) {
     line-height: 1.5;
     border-radius: .2rem;
 }
+
+html:not(.index_page) body {
+    // Ensure UI fills the entire page (and does not run over)
+    > div:first-child {
```
Comment (KKoukiou) [inline pkg/lib/page.scss:555]: This is wrong - will not affected pages which have other div before the main page div, which is very common. Removing it to see what happens.

File: pkg/systemd/terminal.scss [hunk 2]
```diff
@@ -3,10 +3,6 @@
 @import "../../node_modules/@patternfly/patternfly/components/Button/button.css";
 @import "../../node_modules/@patternfly/patternfly/components/Toolbar/toolbar.scss";
 
-html, body {
-        height: 100%;
```
Comment (KKoukiou) [inline pkg/systemd/terminal.scss:7]: terminal page broke - it does not include page.sccs apparently.

**Review by KKoukiou — COMMENTED** [review]
Not all pages include page.scss - this needs to be moved in patternfly-4-overrides probably.

**Comment by garrett** [comment]
## Network
- Spinner is at the top (might be related)
- Blank space at the bottom (I did control-minus to shrink the page): Screenshot_2021-02-03 Networking - garrett Rain

## Diagnostic Reports
- This page is weird... 
Screenshot_2021-02-03 Diagnostic Reports - garrett Rain

## Kernel Dump
- Page is centered vertically; it shouldn't be in this case. But it should be stretched still.
- Also, there's a vertical scrollbar, even with all this space!?
- Screenshot: 
Screenshot_2021-02-03 Kernel Dump - garrett Rain

## Software Updates, ostree edition

- The page doesn't stretch... but this is an external repo (so I'm not sure if it'll inherit from Cockpit... I think it doesn't directly)
   Screenshot_2021-02-03 Software Updates - garrett Rain

## Terminal
- It's broken, but you mentioned that you fixed it. I'm updating now. Perhaps some of the above is out-of-date too.

**Comment by martinpitt** [comment]
Dropping release-blocker. This is too intrusive for a last-minute landing, let's instead just do a pinpointed fix on the updates page, and land this after the release.

File: pkg/lib/patternfly/patternfly-4-overrides.scss [hunk 3]
```diff
@@ -76,3 +76,13 @@ svg {
       }
   }
 }
+
+// Let PF4 handle the scrolling through page component otherwise we might get double scrollbar
+html:not(.index-page) body {
+  overflow-y: hidden;
+
+  // Ensure UI fills the entire page (and does not run over)
+  div.pf-u-h-100 {
```
Comment (garrett) [inline pkg/lib/patternfly/patternfly-4-overrides.scss:85]: 1. This shouldn't be tied to a `<div>` element. We really should not have CSS bound to elements _and_ classes (unless necessary... but it's not necessary ~99% of the time).
2. This doesn't look like a PF class, but a CT one? It's Cockpit-specific, right? Something like ct-page-fill might be better, as it's more descriptive and signifies that it's a Cockpit-specific utility class related to the page.

**Review by garrett — DISMISSED** [review]
We shouldn't have Cockpit specific CSS masquerade as PF CSS.

I haven't done an in-Cockpit review; I just looked at the CSS diff on this new first-pass. I'll look in a bit.

File: pkg/lib/patternfly/patternfly-4-overrides.scss (hunk 3 above)
Comment (KKoukiou) [reply pkg/lib/patternfly/patternfly-4-overrides.scss:85]: It's actually a PF class that I found through their website but I can't find it in the shipped code so I just copied their code. I can rename it of course.

Comment (garrett) [reply pkg/lib/patternfly/patternfly-4-overrides.scss:85]: Aha. Weird that it isn't shipped. Perhaps it's new and isn't in the release of PF we're using?

Comment (KKoukiou) [reply pkg/lib/patternfly/patternfly-4-overrides.scss]: They have some stylesheets only for their website, and I think it's part of these.

**Review by garrett — APPROVED** [review]
Works here! LGTM! :+1: 

I've looked through every page both on my machine and in a VM, and I zoomed way out to make things tiny to check.

The only issue is with cockpit-ostree, but I'm 92% sure that needs to be fixed in another repo. :wink:

**Pull request merged**